test/gtest/ucp/test_ucp_perf.cc

```diff
@@ -358,7 +358,16 @@ UCS_TEST_SKIP_COND_P(test_ucp_perf, envelope, has_transport("self"))
 UCP_INSTANTIATE_TEST_CASE(test_ucp_perf)
 
 
-class test_ucp_wait_mem : public test_ucp_perf {};
+class test_ucp_wait_mem : public test_ucp_perf {
+public:
+    static void get_test_variants(std::vector<ucp_test_variant> &variants)
+    {
+        // Add test instance only if wait_mem is non-trivial for this arch
+        if (ucs_arch_wait_mem != ucs_arch_generic_wait_mem) {
```

Comment:
hacky way :)

Reply:
yeah...this is just asking for a trouble...

Reply:
@shamisp is it better now? changed to check arch

Reply:
yes, thanks